tool/tctl/common/resources/database.go

```diff
@@ -0,0 +1,159 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package resources
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils/clientutils"
+	"github.com/gravitational/teleport/lib/asciitable"
+	"github.com/gravitational/teleport/lib/auth/authclient"
+	"github.com/gravitational/teleport/lib/services"
+	sliceutils "github.com/gravitational/teleport/lib/utils/slices"
+	"github.com/gravitational/teleport/tool/common"
+)
+
+type databaseCollection struct {
+	databases types.Databases
+}
+
+// NewDatabaseCollection creates a [Collection] over the provided databases.
+func NewDatabaseCollection(databases types.Databases) Collection {
+	return &databaseCollection{databases: databases}
+}
+
+func (c *databaseCollection) Resources() []types.Resource {
+	return sliceutils.Map(c.databases, func(db types.Database) types.Resource {
+		return db
+	})
+}
+
+func (c *databaseCollection) WriteText(w io.Writer, verbose bool) error {
+	var rows [][]string
```

Comment:
Suggestion: preallocate since we know how many rows this will have ahead of time.
```suggestion
	rows := make([][]string, 0, len(c.databases))
```